Docusaurus v2

JLHwung opened the pull request:
Reviving https://github.com/babel/website/pull/2622.

Resolves #2622

Todo:
- [x] Setup page migration
- [x] UI breakage on dark mode
- [x] Syntax highlighting theme
- [x] Assumptions page REPL
- [x] Algolia DocSearch index migration
- [x] Redirect v1 URL `/docs/en/*` to v2 `/doc/:splat`
- [x] Cleanup `Todo: strek` items

Known limitations:

- The Docs Search result still refers to v1 URL `/docs/en/...`. When this PR gets merged, I will manually have Agolia re-crawled the website so that the search result can be updated.

harish-sethuraman commented:
Hey 👋🏻 you need some help? It has been a long time since I had revisited my PR as well?

JLHwung commented:
@harish-sethuraman Hi. Sorry for holding that PR long, it's easy to get distracted...

So I am slowly working to migrate the whole website to docusaurus v2 based on your amazing work in #2622. Since then the setup page has been migrated, tools/timeline are working now and syntax highlighting migrated, too. Also fixed a few UI breakage, especially on dark mode.

> you need some help?

Thank you! I have added a todo list and feel free to pick one if you want to help. If you have limited bandwidth, randomly visiting the pages, toggling dark mode / light mode and see if there are any UI issues are also welcome and very helpful. In Docusaurus v1 we don't have dark mode so there are quite a lot hardcoded color fg/bg. They need to be replaced by the variables that will respect light/dark theme.

harish-sethuraman commented:
There are some icons that are coming from the assets folder like this one https://deploy-preview-2711--babel.netlify.app/img/icons/twitter.svg  (in [teams](https://deploy-preview-2711--babel.netlify.app/team) page )Im not sure if we can change the colour of this. Can we rather inline this? or import it as a JSX component? so that we can control the themes?

JLHwung commented on scripts/add-blog-social-image.js:
```diff
@@ -1,35 +0,0 @@
-
-  $("meta[name='twitter:card']").attr("content", "summary_large_image");
-  $("meta[property='og:site']").attr("content", "@babeljs");
-  $("meta[property='og:title']").attr("content", title);
-  $("meta[property='og:description']").attr("content", description);
```
docusaurus v2 can infer these two from the frontmatter.

JLHwung commented on scripts/add-blog-social-image.js:
```diff
@@ -1,35 +0,0 @@
-  const filePath = `${__dirname}/../website/build/babel/blog/${url}.html`;
-  const $ = cheerio.load(fs.readFileSync(filePath));
-
-  $("meta[name='twitter:card']").attr("content", "summary_large_image");
```
docusaurus v2 adds it automatically.

JLHwung commented:
> There are some icons that are coming from the assets folder like this one https://deploy-preview-2711--babel.netlify.app/img/icons/twitter.svg (in [teams](https://deploy-preview-2711--babel.netlify.app/team) page )Im not sure if we can change the colour of this. Can we rather inline this? or import it as a JSX component? so that we can control the themes?

Sure. We can inline the assets here. Since this issue is not introduced by the docusaurus v2 upgrade, I suggest we improve that in another PR.

JLHwung commented on website/static/img/icons/github.svg:
```diff
@@ -1 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 499.36"><path d="M256,6.32c-141.36,0-256,114.61-256,256,0,113.09,73.34,209,175.08,242.9,12.8,2.35,17.47-5.56,17.47-12.34,0-6.08-.22-22.18-.35-43.54C121,464.83,106,415,106,415c-11.64-29.57-28.42-37.45-28.42-37.45C54.31,361.71,79.31,362,79.31,362c25.69,1.81,39.21,26.38,39.21,26.38,22.84,39.12,59.92,27.82,74.5,21.27,2.33-16.54,8.94-27.82,16.25 ...
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 499.36"><path d="M256,6.32c-141.36,0-256,114.61-256,256,0,113.09,73.34,209,175.08,242.9,12.8,2.35,17.47-5.56,17.47-12.34,0-6.08-.22-22.18-.35-43.54C121,464.83,106,415,106,415c-11.64-29.57-28.42-37.45-28.42-37.45C54.31,361.71,79.31,362,79.31,362c25.69,1.81,39.21,26.38,39.21,26.38,22.84,39.12,59.92,27.82,74.5,21.27,2.33-16.54,8.94-27.82,16.25 ...
```
The SVG's fill is changed to `currentColor` so that we can manipulate it by `background-color: currentColor`.

JLHwung commented on website/static/css/team.css:
```diff
@@ -1,120 +1,128 @@
+}
+.media-link.github::before {
+  mask: url('/img/icons/github.svg') no-repeat 50% 50%;
+}
```
Credits to https://codepen.io/noahblon/post/coloring-svgs-in-css-background-images

JLHwung commented on docs/assumptions.md:
````diff
@@ -37,7 +32,7 @@ This can be useful, for example, to iterate DOM collections in older browsers.
 <div is="assumption-repl" data-assumption="arrayLikeIsIterable" data-plugins="transform-destructuring,transform-spread,transform-for-of">
 
-```text assumption-input
+```js
````
We can re enable `js` markup now that we retrieve input code from `.token-line`.

harish-sethuraman commented:
This [change](https://github.com/babel/website/pull/2711/commits/7a07a62f854b32162db43abf74fd10ad85014b75) introduced a bug. The landing page wont have the editor at all. Thats why we had to add ace editor in scripts of docusaurus config.
<img width="1191" alt="Screenshot 2023-02-12 at 4 33 34 PM" src="https://user-images.githubusercontent.com/32865581/218307186-6e27579c-ee0f-4770-9db5-3a3f9987e7f8.png">

nicolo-ribaudo reviewed (approved):
![image](https://user-images.githubusercontent.com/7000710/219357479-b35a501b-bfbd-4d26-a423-bc50fa4fcc21.png)

I don't like this brownish-yellow color for the blog post titles, but it looks good on links with smaller font sizes. Maybe we could change the style of these title links? For example, black with a dotted brownish-yellow underline that becomes brownish-yellow with a solid underline on hover.

The rest looks good!

JLHwung commented on website/src/css/custom.css:
```diff
@@ -0,0 +1,161 @@
+  --ifm-blockquote-border-left-width: 4px;
+}
+[data-theme='dark'] {
+  --ifm-color-primary: #f5da55;
```
This is our current primary color, the Babel yellow.

JLHwung commented on website/src/css/custom.css:
```diff
@@ -0,0 +1,161 @@
+:root {
+  --ifm-color-primary: #9e8400;
```
Compared to our previous primary color `#f5da55: hsv(50, 65.3%, 96.1%)`, the new color boosted the saturation and lowered the value `hsv(50, 100%, 62%)` to increase the contrast to the white background.

liuxingbaoyu commented on _redirects:
```diff
@@ -15,68 +15,68 @@ https://babel.netlify.com/* https://babeljs.io/:splat 301!
-/core-packages/babel-code-frame/    /docs/en/babel-code-frame/
-/core-packages/babylon/             /docs/en/babylon/
-/docs/en/babylon/                   /docs/en/babel-parser
+/docs/usage/                        /docs/usage/
```
This one doesn't seem to be needed anymore?

JLHwung commented:
<img width="790" alt="image" src="https://user-images.githubusercontent.com/3607926/219459517-fb09597f-e02b-448f-9fc0-019aa10b607e.png">

In dark mode:

<img width="780" alt="image" src="https://user-images.githubusercontent.com/3607926/219459854-7811608d-8c4d-4b7b-ab59-476f72746b46.png">


The blog post headings have been toned down. I didn't add the dotted underline because it will conflict with the hover styling.

harish-sethuraman commented:
Im still getting the
```
This page crashed.
ace is not defined
```
error in the landing page. Not sure why but I see the ace editor being fetched. Probably the editor is initialised before the script is fetched?

liuxingbaoyu commented:
> Im still getting the
> 
> ```
> This page crashed.
> ace is not defined
> ```
> 
> error in the landing page. Not sure why but I see the ace editor being fetched. Probably the editor is initialised before the script is fetched?

I also ran into this problem and saw it as a caching issue. Can you reproduce in incognito mode, I can't seem to.

JLHwung commented:
> Im still getting the
> 
> ```
> This page crashed.
> ace is not defined
> ```
> 
> error in the landing page. Not sure why but I see the ace editor being fetched. Probably the editor is initialised before the script is fetched?

This is a bug. We run into a racing issue where `minirepl.js` is executed immediately (imported by the `pages/index.js`) while its dependents could be executed after the page are parsed (`defer: true`).
